### PR TITLE
bug fix destroying container

### PR DESCRIPTION
**Pull request by crgimenes**
Just removing the container is not enough, you must first send the kill signal and then remove the container. This PR does this and validates if the container has actually destroyed, if it has not destroyed the system tries three times with an interval of three seconds between attempts.

### CODE REVIEW

File: provision/docker.go [hunk 1]
```diff
@@ -123,6 +123,27 @@ func FnFindImage(client *docker.Client, imageName string) (image docker.APIImage
...
+// FnFindContainerByID return container by ID
+func FnFindContainerByID(client *docker.Client, ID string) (container docker.APIContainers, err error) {
+	var containers []docker.APIContainers
+	containers, err = client.ListContainers(docker.ListContainersOptions{All: true})
+	if err != nil {
+		return
+	}
+	for _, v := range containers {
+		if v.ID == ID {
+			fmt.Println("ID:", v.ID)
```
Comment (avelino) [inline provision/docker.go]: debuger?

Comment (crgimenes) [reply provision/docker.go]: yes... sorry.

File: gofn.go [hunk 2]
```diff
@@ -111,20 +112,40 @@ func Run(ctx context.Context, buildOpts *provision.BuildOptions, containerOpts *
...
+					err = nil
+				}
+				return
+			}
 		}
+		log.Errorf("unable to kill container %v\n", container.ID)
 		return
 	}
-	log.Errorln("docker client and container is nil")
+	log.Errorf("docker client is %#v\n", client, container)
```
Comment (felipeweb) [inline gofn.go]: fix number of args

File: gofn.go [hunk 3]
```diff
@@ -111,20 +112,40 @@ func Run(ctx context.Context, buildOpts *provision.BuildOptions, containerOpts *
...
+	log.Errorf("docker container is %#v\n", client, container)
```
Comment (felipeweb) [inline gofn.go]: fix number of args

Comment (crgimenes) [reply gofn.go]: done

**Pull request merged**